### PR TITLE
Export variable instances as DSv5 variable-fonts

### PR DESCRIPTION
**Pull request by khaledhosny**
Export Glyphs 3 as DesignSpace v5 variable-fonts, setting their custom parameters is the variable-font lib, mainly using the "public.fontInfo" key.

### CODE REVIEW

File: Lib/glyphsLib/builder/instances.py [hunk 1]
```diff
@@ -56,13 +56,41 @@
...
+        axisSubsets=[RangeAxisSubsetDescriptor(name=axis.name) for axis in ds.axes],
+    )
+
+    # to_ufo_custom_params() wants a UFO, create a dummy one
+    ufo = self.ufo_module.Font()
+    to_ufo_custom_params(self, ufo, instance, set_default_params=False)
+
+    info = {}
+    for attr in fontInfoAttributesVersion3:
+        if value := getattr(ufo.info, attr, None):
```
Comment (anthrotype) [inline Lib/glyphsLib/builder/instances.py:86]: wouldn't this `if value := getattr(ufo.info, attr, None)` branch not trigger if `value == 0`?

File: Lib/glyphsLib/builder/instances.py [hunk 2]
```diff
@@ -56,13 +56,41 @@
...
+
+def _to_designspace_varfont(self, instance):
+    from fontTools.designspaceLib import RangeAxisSubsetDescriptor
+    from fontTools.ufoLib import fontInfoAttributesVersion3
+
+    ds = self.designspace
+    ufo_varfont = self.designspace.addVariableFontDescriptor(
+        name=instance.name,
+        filename=instance.customParameters["fileName"],
+        axisSubsets=[RangeAxisSubsetDescriptor(name=axis.name) for axis in ds.axes],
```
Comment (anthrotype) [inline Lib/glyphsLib/builder/instances.py:77]: this feels redundant, I wish one coudl leave axisSubsets empty to mean "this VF uses all the defined axes", but I suppose that's not how the DS spec wants this.

File: Lib/glyphsLib/builder/instances.py [hunk 3]
```diff
@@ -56,13 +56,41 @@
...
+    info = {}
+    for attr in fontInfoAttributesVersion3:
+        if (value := getattr(ufo.info, attr, None)) is not None:
+            info[attr] = value
+
+    if info:
+        ufo_varfont.lib["public.fontInfo"] = info
+
+    for key in ufo.lib:
+        ufo_varfont.lib[key] = ufo.lib[key]
```
Comment (anthrotype) [inline Lib/glyphsLib/builder/instances.py:93]: I wonder, what could end up here? Any params for which there isn't a corresponding fontinfo key basically? I guess it's fine to store them here, even though they won't be used right know by anybody.

**Review by anthrotype — APPROVED** [review]
Thanks Khaled!

File: Lib/glyphsLib/builder/instances.py (hunk 3 above)
Comment (khaledhosny) [reply Lib/glyphsLib/builder/instances.py:93]: Yes. They might be useful for round-tripping, though this PR does not handle it right now.

**Comment by anthrotype** [comment]
This does not round-trip back to .glyphs, if I understand correctly, right? How hard do you reckon will that be? Maybe we could file an issue as a TODO for later.

**Comment by khaledhosny** [comment]
I don’t use ufo2glyphs myself, but I might look into it later. I’ll file an issue for now.

**Pull request merged**